### DISCUSSION TITLE
PowerShell modify_ script

### DISCUSSION THREAD

**Discussion by jeandeaual** [twpayne/chezmoi#1493] [Q&A] [answered]
I'm having trouble writing a `modify_` script in PowerShell.
Basically I'd like to get the contents of the source file into a variable to then modify it. This can be done in a `modify_*.sh` script as follows:
```bash
var="$(cat <&0)"
```

For PowerShell, I tried doing:
```powershell
$var = while ($line = Read-Host) { $line }
```
The issue is that this seems to also output the file contents at the same time, resulting in the output file containing both the unmodified and modified contents.

Does anyone have a working sample of a `modify_*.ps1` script?

**Comment by bradenhilton** [comment, marked as answer]
See https://stackoverflow.com/a/66550241/11000543 regarding `Read-Host`.

I can't remember off the top of my head if `$input`/`$($input)`/`@($input)` works in the context of a chezmoi `modify_` script. Feel free to try it.

At the moment I'm using this:

```powershell
function Read-FileFromStdin {
    $stdin = [System.Console]::In

    $allLines = @()
    while ($line = $stdin.ReadLine()) {
        $allLines += $line
    } 

    if (-not $allLines) {
        return $null
    }

    return $allLines -join [System.Environment]::NewLine
}

$CurrentFileContents = Read-FileFromStdin
```

You can also just directly read the file contents if you want:

```powershell
function Read-FileFromPath {
    $filePath = "your/path/here"

    if (-not (Test-Path -Path $filePath)) {
        return $null
    }

    $allLines = Get-Content -Path $filePath -ErrorAction 'SilentlyContinue'

    if (-not $allLines) {
        return $null
    }

    return $allLines -join [System.Environment]::NewLine
}

$CurrentFileContents = Read-FileFromPath
```

**Reply by jeandeaual** [reply]
I couldn't use `$input`  but your function works wonders. Thanks!
